R/packages.R

```diff
@@ -38,6 +24,22 @@ get_default_wasm_assets <- function(desc) {
       "i" = "Install a package version matching the WebAssembly version to silence this error."
     ))
   }
+}
+
+get_default_wasm_assets <- function(desc) {
+  pkg <- desc$Package
+  r_wasm <- "http://repo.r-wasm.org"
+  r_short <- gsub("\\.[^.]+$", "", WEBR_R_VERSION)
+  contrib <- glue::glue("{r_wasm}/bin/emscripten/contrib/{r_short}")
+
+  info <- utils::available.packages(contriburl = contrib)
+  if (!pkg %in% rownames(info)) {
+    cli::cli_warn("Can't find {.pkg {pkg}} in webR binary repository.")
+    return(list())
+  }
+
+  ver <- info[pkg, "Version", drop = TRUE]
+  warn_repo_pkg_version(desc, ver)
```

Comment:
Another small nit: I do think it'd be worth refactoring the repeated code of these two functions into a helper function. That would make it easier to see how the two functions differ.

Reply:
OK, I'm happy to be convinced here. I was originally worried about if r-wasm and r-universe diverge again in future but in truth the plan is for that not to happen. Plus we can always just separate things back out if we need to.

Reply:
I've combined the two functions into a single `get_wasm_assets()` function that takes a second argument to handle which repo URL to grab assets from.